pytest/lib/bridge.py

```diff
@@ -223,99 +344,163 @@ def __init__(self, config):
-        print('cwd', self.bridge_dir)
-        print('yarn version')
-        exit_code = os.system('yarn --version')
-        print('exit code', exit_code)
-        print('node version')
-        exit_code = os.system('node --version')
-        print('exit code', exit_code)
-        os.system('which node')
-        assert subprocess.check_output(['yarn'], cwd=self.bridge_dir).decode(
-            'ascii').strip().split('\n')[-1].strip().split(' ')[0] == 'Done'
+        assert_success(subprocess.check_output(['yarn'], cwd=self.bridge_dir))
         ethash_dir = os.path.join(self.bridge_dir, 'eth2near/ethashproof')
-        assert subprocess.check_output(
-            ['/bin/sh', 'build.sh'], cwd=ethash_dir) == b''
+        assert_success(subprocess.check_output(['/bin/sh', 'build.sh'], cwd=ethash_dir))
         # Make sure that ethash is assembled
         assert os.path.exists(os.path.join(ethash_dir, 'cmd/relayer/relayer'))
 
     def init_near_contracts(self):
-        print('Init NEAR contracts...')
-        assert subprocess.check_output(['node', 'index.js', 'init-near-contracts'], cwd=self.cli_dir).decode(
-            'ascii').strip().split('\n')[-1] == 'ETH2NEARProver initialized'
+        assert_success(self.adapter.js_call_cli('init-near-contracts', timeout=60))
 
     def init_eth_contracts(self):
-        print('Init ETH contracts...')
-        assert_deployed(subprocess.check_output(
-            ['node', 'index.js', 'init-eth-ed25519'], cwd=self.cli_dir))
-        assert_deployed(
-            subprocess.check_output(
-                [
-                    'node',
-                    'index.js',
-                    'init-eth-client',
-                    '--eth-client-lock-eth-amount',
-                    '1000000000000000000',
-                    '--eth-client-lock-duration',
-                    '30'],
-                cwd=self.cli_dir))
-        assert_deployed(subprocess.check_output(
-            ['node', 'index.js', 'init-eth-prover'], cwd=self.cli_dir))
-        assert_deployed(subprocess.check_output(
-            ['node', 'index.js', 'init-eth-erc20'], cwd=self.cli_dir))
-        assert_deployed(subprocess.check_output(
-            ['node', 'index.js', 'init-eth-locker'], cwd=self.cli_dir))
-
+        assert_success(self.adapter.js_call_cli('init-eth-ed25519', timeout=60))
+        assert_success(self.adapter.js_call_cli([
+            'init-eth-client',
+            '--eth-client-lock-eth-amount',
+            '1000000000000000000',
+            '--eth-client-lock-duration',
+            '30'], timeout=60))
+        assert_success(self.adapter.js_call_cli('init-eth-prover', timeout=60))
+        assert_success(self.adapter.js_call_cli('init-eth-erc20', timeout=60))
+        assert_success(self.adapter.js_call_cli('init-eth-locker', timeout=60))
+
+    @retry(stop_max_attempt_number=10, wait_fixed=1000)
     def init_near_token_factory(self):
```

Comment:
Why do we need a `retry` on this method? what is the criteria to use `retry`? I see it in some of the methods but not in all that I expect. 

(tbh ideally they won't be used anywhere, but I guess interfaces are not robust enough)

Reply:
You're right, it's about robustness. Here's the exact failure: https://nayduckdiag.blob.core.windows.net/logs/66198_v2_stderr.